src/ert/config/forward_model_step.py

```diff
@@ -46,6 +72,46 @@ class ForwardModelStep:
         "_ERT_RUNPATH": "<RUNPATH>",
     }
 
+    def validate_pre_experiment(self, fm_step_json: ForwardModelStepJSON) -> None:
+        """
+        Raise errors pertaining to the environment not being
+        as the forward model step requires it to be. For example
+        a missing FLOW version.
+        """
+        pass
+
+    def validate_pre_realization_run(
+        self, fm_step_json: ForwardModelStepJSON
+    ) -> ForwardModelStepJSON:
+        """
+        Used to validate/modify the job JSON to be run by the forward model step runner.
+        It will be called for every joblist.json created.
+        """
+        return fm_step_json
+
+    def __post_init__(self) -> None:
+        # We unescape backslash here to keep backwards compatability ie. If
+        # the arglist contains a '\n' we interpret it as a newline.
+        self.arglist = [
+            s.encode("utf-8", "backslashreplace").decode("unicode_escape")
+            for s in self.arglist
+        ]
+
+        self.environment.update(ForwardModelStep.default_env)
+
+        if self.stdout_file is None:
+            self.stdout_file = f"{self.name}.stdout"
+        elif self.stdout_file == "null":
+            self.stdout_file = None
+
+        if self.stderr_file is None:
+            self.stderr_file = f"{self.name}.stderr"
+        elif self.stderr_file == "null":
+            self.stderr_file = None
+
+        if self.stdin_file == "null":
+            self.stdin_file = None
+
```

Comment:
We probably don't want to expose the `from_config_file` classmethod. We can probably move this classmethod to `ert_config.py`.